src/misc/setup.cpp

```diff
@@ -1209,61 +1208,45 @@ void Config::Init() const
 	}
 }
 
-void Section::AddInitFunction(SectionFunction func, bool changeable_at_runtime)
+void Section::AddFunctions(const SectionFunctions *funcs)
 {
-	if (func) {
-		init_functions.emplace_back(func, changeable_at_runtime);
+	if (funcs && (funcs->init || funcs->destroy)) {
+		section_functions.push_back(*funcs);
 	}
 }
 
-void Section::AddDestroyFunction(SectionFunction func, bool changeable_at_runtime)
-{
-	destroyfunctions.emplace_front(func, changeable_at_runtime);
-}
-
 void Section::ExecuteInit(const bool init_all)
 {
-	for (size_t i = 0; i < init_functions.size(); ++i) {
+	for (const auto &funcs : section_functions) {
 		// Can we skip calling this function?
-		if (!(init_all || init_functions[i].changeable_at_runtime)) {
+		if (!funcs.init) {
+			continue;
+		}
+		if (!(init_all || funcs.execute_on_config_change)) {
 			continue;
 		}
 
 		// Track the size of our container because it might grow.
-		const auto size_on_entry = init_functions.size();
-
-		assert(init_functions[i].function);
-		init_functions[i].function(this);
-
-		const auto size_on_exit = init_functions.size();
-
-		if (size_on_exit > size_on_entry) {
-			//
-			// If the above function call appended items then we
-			// need to avoid calling them immediately in this
-			// current pass by advancing our index across them. The
-			// setup class will call the added function itself.
-			//
-			const auto num_appended = size_on_exit - size_on_entry;
-			i += num_appended;
-			assert(i < init_functions.size());
-		}
+		const auto size_on_entry = section_functions.size();
```

Comment:
Add a [[maybe_unused]], since these variables are unused if not compiled in DEBUG-mode.